### PR TITLE
Add copying of stall and review details from user interface

### LINKED ISSUE #148: Add copying of stall and review details from user interface

### PR DESCRIPTION
**Pull request by Hikoya**
Resolve #148

### CODE REVIEW

File: src/main/java/foodwhere/ui/ReviewListPanel.java [hunk 1]
```diff
@@ -30,6 +35,46 @@ public ReviewListPanel(ObservableList<Review> reviewList) {
...
+        final ContextMenu contextMenu = new ContextMenu();
+        MenuItem copyReviewName = new MenuItem("Copy Name");
+        MenuItem copyReviewDate = new MenuItem("Copy Date");
+        MenuItem copyReviewContent = new MenuItem("Copy Content");
+        MenuItem copyReviewTag = new MenuItem("Copy Tag");
+        contextMenu.getItems().addAll(copyReviewName, copyReviewDate, copyReviewContent, copyReviewTag);
+
+        reviewListView.setContextMenu(contextMenu);
+
+        copyReviewName.setOnAction((ActionEvent actionEvent) -> {
```
Comment (clarence-chew) [inline src/main/java/foodwhere/ui/ReviewListPanel.java:57]: Seems quite repetitive - perhaps we can consider abstracting this as a CopyMenuItem<Review> class?

**Review by clarence-chew — APPROVED** [review]
LGTM. I think there are some places to consider reducing repeated code.

File: src/main/java/foodwhere/ui/ReviewListPanel.java (hunk 1 above)
Comment (Hikoya) [reply src/main/java/foodwhere/ui/ReviewListPanel.java]: Thanks for the suggestion, I have abstracted out

**Review by clarence-chew — COMMENTED** [review]
I have some comments here on how we can consider abstracting this.

File: src/main/java/foodwhere/ui/CopyMenuItem.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        super(tag);
+        requireAllNonNull(tag, item, clipboard, action);
+        this.item = item;
+        this.clipboard = clipboard;
+        this.action = action;
+        this.setOnAction();
+    }
+
+    private void setError() {
+        throw new RuntimeException("Invalid type!");
```
Comment (clarence-chew) [inline src/main/java/foodwhere/ui/CopyMenuItem.java:93]: The edits look good, but currently `throw new RuntimeException("Invalid type!");` is repeated - perhaps we should reuse that function?

**Review by clarence-chew — APPROVED** [review]
Code looks clean - LGTM.

File: src/main/java/foodwhere/ui/CopyMenuItem.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    /**
+     * Sets the content of the clipboard according to the type of the item
+     */
+    public void setOnAction() {
```
[the remark this answers is not in the document]
Comment (Hikoya) [reply src/main/java/foodwhere/ui/CopyMenuItem.java:107]: Resolved in latest commits

File: src/main/java/foodwhere/ui/CopyMenuItem.java (hunk 2 above)
Comment (Hikoya) [reply src/main/java/foodwhere/ui/CopyMenuItem.java:101]: Resolved in latest commits

**Review by clarence-chew — APPROVED** [review]
LGTM.

**Pull request merged**